modules/processing/network.py

```diff
@@ -371,6 +410,14 @@ def _add_dns(self, udpdata):
                 # TODO: add srv handling
                 query["answers"].append(ans)
 
+            if ( self.whitelist_enabled and 
+                (conn.get("src") in self.known_dns or conn.get("dst") in self.known_dns) and  
+                q_name in self.whitelist 
+            ):
```

Comment:
refactored the check and it's more dynamic now as moved to a function. Also adds option that if list is empty then it means accept all